### ISSUE TITLE
`tsc --watch` doesn't pick up on changes to `--jsx`

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#51829]
```json5
// ./tsconfig.json
{
    "compilerOptions": {
        "target": "esnext",
        "module": "commonjs",
        "noEmit": true,
    }
}
```

```tsx
// ./src/index.tsx
declare var Foo: any;
let x = <Foo></Foo>;
```

```
[4:24:35 PM] Starting compilation in watch mode...

src/index.tsx:3:9 - error TS17004: Cannot use JSX unless the '--jsx' flag is provided.

3 let x = <Foo></Foo>;
          ~~~~~

[4:24:37 PM] Found 1 error. Watching for file changes.
```

Now add `"jsx": "react",` to the `tsconfig.json`.

```diff
  {
      "compilerOptions": {
+         "jsx": "react",
          "target": "esnext",
          "module": "commonjs",
          "noEmit": true,
      }
  }
```

**Expected**: Error should be gone.
**Actual**: Error is still present.

**Comment by xgqfrms** [comment]
any update?

**Comment by jakebailey** [comment]
Hm, can you try the beta or nightly? I think this was fixed in #53403 and this issue is an accidental dupe of #53391.
